src/lib/ui/AccessCodeTable/AccessCodeTable.tsx

```diff
@@ -8,17 +6,23 @@ import {
   TableTitle,
 } from 'lib/ui/Table/index.js'
 
+import { useAccessCodes } from 'lib/hooks/use-access-codes.js'
 import { AccessCodeKeyIcon } from 'lib/icons/AccessCodeKey.js'
 import { CodeDetails } from 'lib/ui/AccessCodeTable/CodeDetails.js'
 import { ContentHeader } from 'lib/ui/layout/ContentHeader.js'
 import { Caption } from 'lib/ui/typography/Caption.js'
 import { Title } from 'lib/ui/typography/Title.js'
 
 export function AccessCodeTable(props: {
-  accessCodes: AccessCode[]
+  deviceId: string
   onClickBack?: () => void
 }): JSX.Element {
-  const { accessCodes, onClickBack } = props
+  const { accessCodes } = useAccessCodes({
+    device_id: props.deviceId,
+  })
+  const { onClickBack } = props
+
+  if (!accessCodes) return <>{null}</>
```

Comment:
Interesting, I think this caused a type error with `JSX.Element`. Can also fix via `return null` and updating type to `JSX.Element | null`